spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryClient.java

```diff
@@ -67,13 +68,10 @@ public String description() {
 
 	private void addInstancesToList(List<ServiceInstance> instances, String serviceId, QueryParams queryParams) {
 		HealthServicesRequest.Builder requestBuilder = HealthServicesRequest.newBuilder()
-				.setPassing(this.properties.isQueryPassing()).setQueryParams(queryParams)
-				.setToken(this.properties.getAclToken());
-		String queryTag = this.properties.getQueryTagForService(serviceId);
-		if (queryTag != null) {
-			requestBuilder.setTag(queryTag);
-		}
+				.setPassing(properties.isQueryPassing()).setQueryParams(queryParams).setToken(properties.getAclToken());
+		Optional.ofNullable(properties.getQueryTagsForService(serviceId)).ifPresent(requestBuilder::setTags);
```

Comment:
The new `getQueryTagsForService` behaves identical to the previously used `getQueryTagForService` in the single tag value case. *IFF* there are multiple tag property values does it diverge (by adding multiple tags).

Reply:
In Spring Cloud we don't use optional as an if statement at all